### DISCUSSION TITLE
Why does react query in react native have the same default refetch behavior as in the browser?

### DISCUSSION THREAD

**Discussion by ianjkaplan** [TanStack/query#3597] [Ideas]
I see that the docs show how to nicely implement the sensible react query defaults available in the browser [here](https://react-query.tanstack.com/react-native#refresh-on-screen-focus). 

Are there good reasons not to build this behavior into the library? What are some potential issues with detecting the platform
and implementing the same behavior on setup so to remove this responsibility from application code since we would want this aggressive synchonization enabled by default? Is the main issue the reliance on react-native specific dependencies?

Thanks in advance!

**Comment by TkDodo** [comment]
mainly we are not adding 3rd party dependencies and try not to build platform specific things into the library. Some things like window focus refetching favour the browser, but are built in a way so that they can be extended for other platforms (focusManager, onlineManager)
